### DISCUSSION TITLE
Utiliser API pour le téléchargement ?

### DISCUSSION THREAD

**Discussion by pbda** [ssenart/PyGazpar#31] [Ideas]
Bonjour,
Ne serait-il pas plus simple et moins sujet au modifications des pages web, après la connexion d'appeler directement le téléchargement des index après la connexion ? En manuel, ça marche avec une URL de type:

`https://monespace.grdf.fr/api/e-conso/pce/consommation/informatives/telecharger?dateDebut=2021-10-24&dateFin=2021-11-24&frequence=Journalier&pceList%5B%5D=[n°PCE]`

Vue dans le projet [domoticz_gazpar](https://github.com/empierre/domoticz_gaspar/issues/29)

**Comment by ssenart** [comment]
Bonjour,

Oui, bien vu, le nouveau site permet cela. J'avoue que je n'ai pas creusé depuis de la mise à jour du site la semaine dernière.
En effet, cela va simplifier énormément l'affaire, puisqu'on va pouvoir se passer de Selenium.

Bon, j'attend un peu que le site se stabilise, car pour l'instant, mes données ne sont plus mises à jour depuis plus d'une semaine.
En plus, il y a la menace qu'un CAPTCHA soit mise en place (je l'ai déjà vu, mais ils l'ont apparemment retiré depuis) et cela mettra fin à tout possibilité de récupérer les données automatiquement.

En tout cas, merci pour ton message.

**Comment by pbranly** [comment]
Hello
Comme dit sur le GitHub d’un autre système , les valeurs sur GRDF s’arrêtent le 23. Alors que EDF chez qui je suis abonné aussi au gaz, donne des valeurs beaucoup plus récentes .
Donc EDF a possibilité de récupérer des valeurs ailleurs 
Ça c’est plus inquiétant ( en plus du captcha potentiel)
Phil

**Comment by pbda** [comment]
EDF doit utiliser l'API ADICT de GRDF. Il faudrait faire un système avec un tiers professionnel comme [DomoticzLinky](https://github.com/guillaumezin/DomoticzLinky) le fait avec l'aide de [Russandol.pro](https://russandol.pro/). Ça permettrait d'utiliser l'API ADICT de GRDF.

**Comment by ssenart** [comment]
Hello, 

La nouvelle pseudo API présenté par [pbda](https://github.com/pbda)  est maintenant implémenté dans la version 1.1.1. 

Résultat : Plus de Selenium, ni de geckodriver, ni de Firefox, c'est plus léger et beaucoup plus rapide.

Merci encore pbda !

En fouillant, j'ai vu qu'on peut aussi interroger une URL et avoir directement les données en JSON plutôt qu'en Excel. Peut-être qu'on peut encore faire mieux...

Pour ceux qui sont sous Home Assistant, j'ai aussi mis à jour l'intégration Gazpar ici : https://github.com/ssenart/ha-custom_components

**Comment by pbda** [comment]
Merci. As-tu trouvé une doc sur cette API de la nouvelle version ?

**Comment by ssenart** [comment]
Non, je n'ai pas de doc.
